src/Tvheadend.cpp

```diff
@@ -482,10 +483,21 @@ PVR_ERROR CTvheadend::GetRecordings ( ADDON_HANDLE handle )
         auto cit = m_channels.find(rec.iChannelUid);
         if (cit == m_channels.cend())
           rec.channelType = PVR_RECORDING_CHANNEL_TYPE_UNKNOWN;
-        else if (cit->second.IsRadio())
-          rec.channelType = PVR_RECORDING_CHANNEL_TYPE_RADIO;
         else
-          rec.channelType = PVR_RECORDING_CHANNEL_TYPE_TV;
+        {
+          switch (cit->second.GetType())
+          {
+            case CHANNEL_TYPE_TV:
+              rec.channelType = PVR_RECORDING_CHANNEL_TYPE_TV;
+              break;
+            case CHANNEL_TYPE_RADIO:
+              rec.channelType = PVR_RECORDING_CHANNEL_TYPE_RADIO;
+              break;
+            default:
```

Comment:
For the sake of completeness I would also handle CHANNEL_TYPE_OTHER explicitely here.